src/vm/analysis/type_checker/tests/mod.rs

```diff
@@ -54,6 +54,40 @@ fn test_get_block_info(){
     }
 }
 
+#[test]
+fn test_stx_ops(){
+    let good = ["(stx-burn? u10 'SM2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQVX8X0G)",
+                "(stx-transfer? u10 tx-sender 'SM2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQVX8X0G)"];
+    let expected = [ "(response bool uint)", "(response bool uint)" ];
+
+    let bad = [
+        "(stx-transfer? u4 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR)",
+        "(stx-transfer? 4 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR)",
+        "(stx-transfer? u4 u3  'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR)",
+        "(stx-transfer? u4 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR 'true)",
+        "(stx-burn? u4)",
+        "(stx-burn? 4 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR)",
+        "(stx-burn? u4 'true)",
+        "(stx-burn? u4 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR)",
+    ];
+    let bad_expected = [ CheckErrors::IncorrectArgumentCount(3,2),
```

Comment:
Can we also have some tests covering transfers from:
- users to contracts,
- contracts to contracts, 
- contracts to users
?

Reply:
Yep, will add

Reply:
Added in 1063f9f0d